### PR TITLE
Commits for the art gallery application

### PR DESCRIPTION
**Pull request by meghanasharathd**
Description
This pull request contains all the required files for a React based art gallery application. 

Components Added

1. Home.jsx
This React component, named "Home", serves as the homepage for an art gallery website. It features a carousel display showcasing artworks fetched from a service using the useEffect hook upon component mount. The component utilizes React Router for navigation and the react-responsive-carousel library to render the carousel. It gracefully handles loading states by displaying a loading animation while fetching data and switches to displaying the carousel once the data is loaded. Each carousel item links to a detailed information page about the corresponding artwork.
2.InfoPiece.jsx
This React component, "InfoPiece", retrieves and displays detailed information about a specific artwork using the objectNumber parameter from the URL. It fetches data upon component mount or parameter change, displaying loading indicators and error messages as necessary, before presenting the artwork's title, image, artist, year, and description.
3. Login.jsx
This React component, "Login", provides a user interface for logging in. It includes form fields for email and password, handles user input using useState, and triggers a login action upon form submission. Upon successful login, it displays a success message. Styling is enhanced with Bootstrap components and custom CSS.
4. SearchArtWorks.jsx
"SearchArtworks" is a React component facilitating artwork search by title or artist. It utilizes useState for query management, fetching data from an API service upon form submission. Results are displayed in cards, with loading indicators and error messages for enhanced user experience.
5.Signup.jsx 
Presents a signup form

### CODE REVIEW

File: src/services/apiService.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// apiService.js
+// This service module handles all API interactions for the Art Gallery application,
+// encapsulating the logic needed to fetch data from the Rijksmuseum API.
+
+import axios from "axios";
+
+const API_URL = "https://www.rijksmuseum.nl/api/en/collection";
+const apiKey = "KntueOG1";
```
Comment (SiddharthSubash) [inline src/services/apiService.js]: Used hardcoded api key here

File: vite.config.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { defineConfig } from 'vite'
+import react from '@vitejs/plugin-react'
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react()],
+})
```
Comment (SiddharthSubash) [inline vite.config.js:7]: Code functionality is working as expected.

Correctly adhered to coding styling guidelines and bootstrap react.
Overall code structure is well-defined and easily maintainable and readable.

File: src/components/SearchArtworks.jsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      setLoading(false); 
+    }
+  };
+
+  return (
+    <Container>
+      <Form onSubmit={handleSearch} className="mb-4">
+        {/* Label for search form */}
+        <Form.Group controlId="searchQuery">
+          <Form.Label>Search Artworks by Title or Artist</Form.Label>
```
Comment (SiddharthSubash) [inline src/components/SearchArtworks.jsx]: The heading is a bit greyed out. It is not clearly visible

File: src/components/Signup.jsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+              </Button>
+            </Form>
+          </div>
+        </Col>
+      </Row>
+    </Container>
+  );
+};
+
+export default Signup;
```
Comment (SiddharthSubash) [inline src/components/Signup.jsx:94]: Coding style guideline is followed and easy to maintain and better readability.

File: src/services/apiService.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    );
+    return response.data.artObjects;
+  } catch (error) {
+    console.error("Error fetching artworks by maker: ", error);
+    return [];
+  }
+};
+
+export const getPieceById = async (id) => {
+  //const url = `${API_URL}/${objectNumber}?key=${apiKey}`;
```
Comment (SiddharthSubash) [inline src/services/apiService.js]: Unwanted comment line here.

File: src/context/GalleryContext.jsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  return (
+    <GalleryContext.Provider value={{ data, fetchData }}>
+      {children}
+    </GalleryContext.Provider>
+  );
+};
```
Comment (SiddharthSubash) [inline src/context/GalleryContext.jsx:24]: It is good that you have used a context here along with use effect.

File: src/services/apiService.js (hunk 1 above)
Comment (adtsim) [reply src/services/apiService.js]: +1

File: src/components/SearchArtworks.jsx (hunk 3 above)
Comment (adtsim) [reply src/components/SearchArtworks.jsx]: +1

File: src/components/Home.jsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import apiService from "../services/apiService";
```
Comment (adtsim) [inline src/components/Home.jsx:3]: Effectively separating concerns by using a service layer (apiService) to handle API calls is a good approach, this makes the code cleaner and also more maintainable and easier to update or expand.

File: src/components/SearchArtworks.jsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+            onChange={(e) => setQuery(e.target.value)}
+            className="mb-2"
+          >
+            <option value="">Select an artist</option>
+            {artists.map((artist, index) => (
+              <option key={index} value={artist}>{artist}</option>
+            ))}
+          </Form.Control>
+          {/* Input field to enter search query */}
+          <Form.Control
```
Comment (adtsim) [inline src/components/SearchArtworks.jsx]: The Form.Group component uses controlId="searchQuery", which should match an existing element's ID for labels to be properly associated with form controls. You could add a form control (like an input) with the id searchQuery.

Comment (meghanasharathd) [reply src/components/SearchArtworks.jsx]: Added as suggested!

File: src/services/apiService.js (hunk 5 above)
Comment (meghanasharathd) [reply src/services/apiService.js]: Removed

**Review by adtsim — APPROVED** [review]

**Review by SiddharthSubash — APPROVED** [review]

**Pull request merged**